### PR TITLE
[Distributed][Test] Fix todo in distributed test files

### PR DESCRIPTION
**Pull request by zeshengzong**
Refactor distributed test code:
- Fix TODO: (rohan-varma): remove model
- Fix TODO: add comments for TestTraverse
- Migrate deprecated method call `load_state_dict` and `save_state_dict`

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @LucasLLC @MeetVadakkanchery @mhorowitz @pradeepfn

### CODE REVIEW

File: test/distributed/fsdp/test_distributed_checkpoint.py [hunk 1]
```diff
@@ -7,8 +7,8 @@
 from torch.distributed.checkpoint import (
     FileSystemReader,
     FileSystemWriter,
-    load_state_dict,
-    save_state_dict,
+    load,
+    save,
```
Comment (kwen2501) [inline test/distributed/fsdp/test_distributed_checkpoint.py:11]: is this a name change or is there functionality change?
cc @fegin

**Review by kwen2501 — APPROVED** [review]
Looks reasonable to me. Thanks!
Left a comment.

**Comment by zeshengzong** [comment]
@kwen2501 Hello, fixed a lint problem, please review the change when available, thanks!

File: test/distributed/fsdp/test_distributed_checkpoint.py (hunk 1 above)
Comment (wz337) [reply test/distributed/fsdp/test_distributed_checkpoint.py]: Mostly a name change and API signature change.

**Comment by zeshengzong** [comment]
@wz337 Hi, need help to trigger CI I think, thanks! :D

**Pull request closed without merging**